temporalio/contrib/openai_agents/_temporal_openai_agents.py

```diff
@@ -172,226 +174,89 @@ def __init__(self) -> None:
+            raise ValueError("No WorkflowRunner provided to the OpenAI plugin.")
 
-        server_names = [server.name for server in self._mcp_server_providers]
-        if len(server_names) != len(set(server_names)):
-            raise ValueError(
-                f"More than one mcp server registered with the same name. Please provide unique names."
-            )
-
-        for mcp_server in self._mcp_server_providers:
-            new_activities.extend(mcp_server._get_activities())
-        config["activities"] = list(config.get("activities") or []) + new_activities
-
-        runner = config.get("workflow_runner")
+        # If in sandbox, add additional passthrough
         if isinstance(runner, SandboxedWorkflowRunner):
-            config["workflow_runner"] = dataclasses.replace(
+            return dataclasses.replace(
                 runner,
                 restrictions=runner.restrictions.with_passthrough_modules("mcp"),
             )
-
-        config["workflow_failure_exception_types"] = list(
-            config.get("workflow_failure_exception_types") or []
-        ) + [AgentsWorkflowError]
-        return self.next_worker_plugin.configure_worker(config)
-
-    async def run_worker(self, worker: Worker) -> None:
-        """Run the worker with OpenAI agents temporal overrides.
-
-        This method sets up the necessary runtime overrides for OpenAI agents
-        to work within the Temporal worker context, including custom runners
-        and trace providers.
-
-        Args:
-            worker: The worker instance to run.
-        """
-        with set_open_ai_agent_temporal_overrides(self._model_params):
-            await self.next_worker_plugin.run_worker(worker)
-
-    def configure_replayer(self, config: ReplayerConfig) -> ReplayerConfig:
-        """Configure the replayer for OpenAI Agents."""
-        config["interceptors"] = list(config.get("interceptors") or []) + [
-            OpenAIAgentsTracingInterceptor()
-        ]
-        config["data_converter"] = self._data_converter(config.get("data_converter"))
-        return self.next_worker_plugin.configure_replayer(config)
+        return runner
 
     @asynccontextmanager
-    async def run_replayer(
-        self,
-        replayer: Replayer,
-        histories: AsyncIterator[temporalio.client.WorkflowHistory],
-    ) -> AsyncIterator[AsyncIterator[WorkflowReplayResult]]:
-        """Set the OpenAI Overrides during replay"""
-        with set_open_ai_agent_temporal_overrides(self._model_params):
-            async with self.next_worker_plugin.run_replayer(
-                replayer, histories
-            ) as results:
-                yield results
+    async def run_context() -> AsyncIterator[None]:
+        with set_open_ai_agent_temporal_overrides(model_params):
+            yield
+
+    return create_plugin(
+        name="OpenAIAgentsPlugin",
+        data_converter=_data_converter,
+        worker_interceptors=[OpenAIAgentsTracingInterceptor()],
```

Comment:
If, say, I as a plugin author needed an interceptor only applied to workers not replayers. This is a very reasonable thing to do, e.g. an activity interceptor that requires a DB client, which would make no sense in a replayer. What would I as a plugin author have to do here?